### PR TITLE
Update AIESerializer for end-to-end example

### PR DESCRIPTION
**Pull request by frafranz**
With these changes, the pass pipeline for the matmul example (4-core, pack and peel) that was previously defined in the transform dialect can now be run using only C++ passes defined in this repository.

The output TIR is identical to the one obtained before, the commands to use are as follows.

**Compilation to mlir:**

`tools/iree-compile --iree-hal-target-backends=amd-aie --compile-to=executable-sources ../iree-amd-aie/tests/samples/pack_pipeline_funcIR.mlir | tools/iree-opt --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-hal-translate-target-executable-variants{target=amd-aie})))" --iree-amdaie-use-pipeline=pack &> ../incubator-tvm/apps/aie/asr/iree_results.mlir`

**Translation to TIR:**
`./tools/iree-aie-translate --serialize-accel --allow-unregistered-dialect ../incubator-tvm/apps/aie/asr/iree_results.mlir -o ../incubator-tvm/apps/aie/asr/matmul_16x256x256.txt`

Update: Also the tests are consistent with the recent changes now.

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ static LogicalResult setRootConfig(func::FuncOp entryPointFn,
         entryPointFn, matmulOp, tileSizes,
         IREE::Codegen::DispatchLoweringPassPipeline::None);
   } else if (usePassPipeline == AIEPassPipeline::PackPipeline) {
-    SmallVector<int64_t> TileSizeLevel0 = {16, 64};
+    SmallVector<int64_t> TileSizeLevel0 = {16, 256};
```
Comment (yzhang93) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp:57]: Please try to add a separate flag to switch between single core and multicore

**Comment by yzhang93** [comment]
Thanks for making the corresponding changes in translator! Now my PR is merged, please rebase.

**Comment by frafranz** [comment]
> Thanks for making the corresponding changes in translator! Now my PR is merged, please rebase.

Great, I just rebased.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp (hunk 1 above)
Comment (frafranz) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp:57]: You're right, I will work out a solution for that.

Comment (yzhang93) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp:57]: To merge this PR, you can change it back to 64 and add another PR for multicore case.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp [hunk 2]
```diff
@@ -284,7 +283,7 @@ void buildAMDAIETransformPassPipeline(OpPassManager &pm) {
 
   if (clUsePipeline != AIEPassPipeline::PackPipeline) {
     auto &modulePassManager = pm.nest<ModuleOp>();
-    addMLIRAIRAIELoweringPasses(modulePassManager);
+      addMLIRAIRAIELoweringPasses(modulePassManager);
```
Comment (yzhang93) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:286]: I think this file should not be changed?

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp (hunk 1 above)
Comment (yzhang93) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp:57]: Oh, I see you also change the test. Then that should be fine. But it's better we can add both single and multicore cases and both tests.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp (hunk 2 above)
Comment (frafranz) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:286]: Oh yes, that was a quirk of my rebasing. Will change it back

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp (hunk 1 above)
Comment (frafranz) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp]: Just pushed the changes, from my side this PR is ready to land!

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp (hunk 2 above)
Comment (frafranz) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp]: done

File: compiler/plugins/target/AMD-AIE/Translation/test/single_core.mlir [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          scf.for %arg2 = %c1 to %c4 step %c2 {
+            scf.forall (%arg3, %arg4) in (%c1, %c3) {
+            }
+          }
+        }
+        return
+      }
+    }
+  }
+}
```
Comment (yzhang93) [inline compiler/plugins/target/AMD-AIE/Translation/test/single_core.mlir:110]: Just a minor thing, then you can merge this PR. Here you should add a new line.

**Review by yzhang93 — APPROVED** [review]

File: compiler/plugins/target/AMD-AIE/Translation/test/single_core.mlir (hunk 3 above)
Comment (frafranz) [reply compiler/plugins/target/AMD-AIE/Translation/test/single_core.mlir]: Done

**Pull request merged**